src/ls.js

```diff
@@ -162,6 +166,18 @@ function _ls(options, paths) {
   list.toEnd = function(file) {
     common.wrap('toEnd', _toEnd, {idx: 1}).call(this.stdout, file);
   };
+  list.cat = function() {
+    return common.wrap('cat', _cat, {idx: 1}).apply(this.stdout, arguments);
```

Comment:
Two things:
1. See my comment above about re-wrapping commands.
2. Idea: What if we turn `ShellString` into something like this:

```javascript
function ShellString(that, err) {
    that = new String(that);
    return ShellStringify(that, that, err);
}

function ShellStringify(that, stdout, stderr) {
    that.stdout = stdout || that.toString();
    that.stderr = stderr || '';
    ['toEnd', 'end', 'cat', 'sed', 'grep', 'exec'].forEach(function(method) {
        that[method] = shell[method];
    });
    return that;
}
```

Then we can use it like this:

```javascript
function _ls(...) {
    // ...
    var list = [...];
    return common.ShellStringify(list, list.join('\n'));
}

function _doSomethingElse(...) {
    // ...
    var list = [...];
    return new common.ShellString('foo');
}

function _somethingBadHappened(...) {
    // ...
    var list = [...];
    return new common.ShellString(null, null, 'everything\'s broken!');
}
```